packages/vsx-registry/src/browser/vsx-extension.tsx

```diff
@@ -351,13 +355,25 @@ export class VSXExtensionComponent extends AbstractVSXExtensionComponent {
 }
 
 export class VSXExtensionEditorComponent extends AbstractVSXExtensionComponent {
+    protected resolveDeferredOnRender(
+        element: HTMLDivElement | null,
+        deferred: Deferred<HTMLDivElement> | undefined): void {
+        if (deferred && element) {
+            deferred.resolve(element);
+            if (element.className === 'body') {
+                this.body = (element || undefined);
+            }
+        }
+    }
+
     render(): React.ReactNode {
         const {
-            builtin, preview, id, iconUrl, publisher, displayName, description,
-            averageRating, downloadCount, repository, license, readme, version
+            builtin, preview, id, iconUrl, publisher, displayName, description, version,
+            averageRating, downloadCount, repository, license, readme, deferredHeaderElementRender,
+            deferredScrollContainerRender, deferredBodyElementRender
         } = this.props.extension;
         return <React.Fragment>
-            <div className='header'>
+            <div className='header' ref={element => this.resolveDeferredOnRender(element, deferredHeaderElementRender)}>
```

Comment:
you should just update the property of this class without any deferred the same in other places

Reply:
I've updated this in my latest changes. The only remaining deferred item is the scroll container which is used for `getScrollContainer` inside `vsx-extension-editor`